openff/evaluator/_tests/test_properties/test_enthalpy.py

```diff
@@ -0,0 +1,354 @@
+                    # keep the output JSON files to pass in as input to mole fraction protocols
+                    parent_outputs.append(output)
+
+            for i in range(n_components):
+                # generate fake data for the conditional groups
+                base = f"extract_observable_component_{i}"
+                path = (
+                    here
+                    / f"{uuid}_conditional_group_component_{i}"
+                    / f"{uuid}_{base}"
+                    / f"{uuid}|{base}.json"
+                )
+                path.parent.mkdir(parents=True, exist_ok=True)
+                name = f"extract_observable_component_{i}"
+                observable = _generate_dummy_observable(name)
+                with path.open("w") as file:
+                    file.write(
+                        json.dumps(
+                            {
+                                ".value": observable.value,
+                                ".time_series_statistics": observable.time_series_statistics,
+                            },
+                            cls=TypedJSONEncoder,
+                        )
+                    )
+                parent_outputs.append((f"{uuid}|{base}", str(path)))
+
+                # now check mole fraction protocol execution
+                cg = workflow_graph.protocols[f"{uuid}|conditional_group_component_{i}"]
+                wmf = cg.protocols[f"{uuid}|weight_by_mole_fraction_{i}"]
+                wmf_path = (
+                    here
+                    / f"{uuid}_conditional_group_component_{i}"
+                    / f"{uuid}_weight_by_mole_fraction_{i}"
+                )
+                protocol_graph._execute_protocol(
+                    str(wmf_path),
+                    wmf,
+                    True,
+                    *parent_outputs,
+                    available_resources=None,
+                    safe_exceptions=True,
+                )
+
+                # check mole fraction substance directly
+                assert np.isclose(
+                    wmf.full_substance.amounts[possible_smiles[i] + "{solv}"][0].value,
+                    output_mole_fractions[i],
+                )
+
+                # check weighted value
+                assert np.isclose(
+                    wmf.weighted_value.value.m_as(unit.kilojoules_per_mole),
+                    output_mole_fractions[i] * 10,
+                )
+
+    def test_expected_output_from_production_simulation(self, tmpdir):
+        """
+        This is an integration test of sorts,
+        constructed to test expected mole fractions. See Issue #575.
+
+        This test steps through the EnthalpyOfMixing process.
+        Some data is provided to avoid re-simulating long trajectories:
+
+        - energy minimisations and equilibrations
+        - the production simulation observables
+        - the decorrelated trajectory and observables
+
+        Note: the test is expected to take a few minutes to run.
```

Comment:
We could move these to GPUs at some point if this is too slow for CI

Reply:
Sounds good -- it is doable right now but definitely one of the slower tests in the codebase.